src/dice/add_dice/randomizer.rb

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+class AddDice
+  class Randomizer
+    attr_reader :dicebot, :cmp_op, :dice_list, :sides
+
+    def initialize(bcdice, dicebot, cmp_op)
+      @bcdice = bcdice
+      @dicebot = dicebot
+      @cmp_op = cmp_op
+      @sides = 0
+      @dice_list = []
+    end
+
+    def roll(times, sides, critical)
+      total = 0
+      results_list = []
+
+      loop_count = 0
+      queue = [times]
+      while !queue.empty?
```

Comment:
```suggestion
      until queue.empty?
```

`while !cond` を `until cond` に書き換えると、否定演算子がなくなって読みやすくなります。

参考：https://techracho.bpsinc.jp/hachi8833/2017_01_10/32333#2-27

Reply:
untilを使うのは個人的に違和感があるため、意図的にwhileを使っていたのですが、やはりuntilの方が読みやすいのでしょうか？ ただでさえ `while` を使うことがRuby的には異質なのに、 `until` というRuby特有な条件式を使うことは読み手に優しくないと思ってました。

理想的には、 `while` すら使わないようにコードを改修したいと思っています。

rubocopのルールは現在オフにしています
https://github.com/bcdice/BCDice/blob/master/.rubocop.yml#L129-L130

Reply:
好みの問題ではあるので、意図的ならばこれで大丈夫です。余談ですが、`until` はPerl由来ではないかと思います。